tb_plugin/tensorboard_plugin_torch_profiler/profiler/module_parser.py

```diff
@@ -235,9 +243,16 @@ def build_node(node, event):
                             logger.warning(
                                 "Runtime and Device-op have same correlation id but with different external id!"
                             )
-                if not tid in tid2list:
-                    tid2list[tid] = []
-                tid2list[tid].append(rt_node)
+                if rt_node.external_id in externalid_to_runtime:
+                    externalid_to_runtime[rt_node.external_id].append(rt_node)
+                else:
+                    externalid_to_runtime[rt_node.external_id] = [rt_node]
```

Comment:
Is this external_id unique in multi-threads scenario?

Reply:
Good question. I double-checked the pytorch code. Now the external id is a static variable inside a function, so it is shared by multi-threads. Because it is atomic type, it is thread safe.
![image](https://user-images.githubusercontent.com/10429114/105845334-92e51b00-6015-11eb-8ea7-1fb3f2409461.png)
But as we see the above TODO, it seems they have plan to make it not unique. But making it not unique will make kernel ambiguous to attach to operator in which thread. I'll contact Gisle to confirm this.

Reply:
That's right - as the comment says we might change the implementation but will still be unique within a single process.